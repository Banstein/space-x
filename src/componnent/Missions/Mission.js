import React from 'react';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { joinMissionDispatcher, leaveMissionDispatcher } from '../../redux/missions/missions';

export default function Mission({ mission }) {
  const dispatch = useDispatch();

  const handleJoinMission = (id) => {
    dispatch(joinMissionDispatcher(id));
  };
  const handleLeaveMission = (id) => {
    dispatch(leaveMissionDispatcher(id));
  };

  const status = mission.status ? (
    <div className="active-mb">Active Member</div>
  ) : (
    <div className="no-active-mb"> NOT A MEMBER</div>
  );

  const missionBt = !mission.status ? (
    <button
      type="button"
      className="join-mission-bt"
      onClick={() => handleJoinMission(mission.missionId)}
    >
      Join Mission
    </button>
  ) : (
    <button
      type="button"
      className="leave-mission-bt"
      onClick={() => handleLeaveMission(mission.missionId)}
    >
      Leave Mission
    </button>
  );

  return (
    <div className="mission-container">
      <div className="mission-name">{mission.missionName}</div>
      <div className="mission-description">{mission.description}</div>
      {status}
      {missionBt}
    </div>
  );
}

Mission.propTypes = {
  mission: PropTypes.oneOfType([PropTypes.object]).isRequired,
};
