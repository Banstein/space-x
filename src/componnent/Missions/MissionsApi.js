import axios from 'axios';

const missionAPI = 'https://api.spacexdata.com/v3/missions';

const getMissionsFromAPI = async () => {
  const getData = await axios.get(missionAPI);
  const missions = [];
  getData.data.forEach((mission) => {
    missions.push({
      missionId: mission.mission_id,
      missionName: mission.mission_name,
      description: mission.mission_description,
    });
  });
  return missions;
};

export default getMissionsFromAPI;
