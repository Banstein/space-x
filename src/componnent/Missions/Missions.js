import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getMissionsDispatcher } from '../../redux/missions/missions';
import Mission from './Mission';
import '../../CSS/base/missions.css';

export default function Missions() {
  const dispatch = useDispatch();
  const isMissionsStored = useSelector((state) => state.missionsReducer.isMissionsStored || false);
  const missions = useSelector((state) => state.missionsReducer.missions || []);

  useEffect(() => {
    if (!isMissionsStored) {
      dispatch(getMissionsDispatcher());
    }
  });

  return (
    <div className="missions-container">
      <div className="header-missions">
        <div className="mission-name-header">Mission</div>
        <div className="mission-description-header">Description</div>
        <div className="mission-status-header">Status</div>
        <div className="mission-other-header">#</div>
      </div>
      {missions.map((mission) => (
        <Mission key={mission.missionId} mission={mission} />
      ))}
    </div>
  );
}
