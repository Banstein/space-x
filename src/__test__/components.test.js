import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Missions from '../componnent/Missions/Missions';
import Mission from '../componnent/Missions/Mission';

it('Test if Missions renders correctly', () => {
  const tree = TestRenderer.create(
    <Provider store={store}>
      <Missions />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});

it('Test if Mission renders correctly', () => {
  const mission = {
    missionId: 1,
    missionName: 'test mission name',
    description: 'description',
  };
  const tree = TestRenderer.create(
    <Provider store={store}>
      <Mission mission={mission} />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
