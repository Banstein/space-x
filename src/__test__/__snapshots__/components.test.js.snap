// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if Mission renders correctly 1`] = `
<div
  className="mission-container"
>
  <div
    className="mission-name"
  >
    test mission name
  </div>
  <div
    className="mission-description"
  >
    description
  </div>
  <div
    className="no-active-mb"
  >
    <span>
       NOT A MEMBER
    </span>
  </div>
  <div
    className="mission"
  >
    <button
      className="join-mission-bt"
      onClick={[Function]}
      type="button"
    >
      Join Mission
    </button>
  </div>
</div>
`;

exports[`Test if Missions renders correctly 1`] = `
<div
  className="missions-container"
>
  <div
    className="header-missions"
  >
    <div
      className="mission-name-header"
    >
      Mission
    </div>
    <div
      className="mission-description-header"
    >
      Description
    </div>
    <div
      className="mission-status-header"
    >
      Status
    </div>
    <div
      className="mission-other-header"
    >
      #
    </div>
  </div>
</div>
`;
